app/src/androidTest/java/com/github/se/eventradar/endtoend/UserFlowTest.kt

```diff
@@ -0,0 +1,489 @@
+import com.github.se.eventradar.model.User
+import com.github.se.eventradar.model.event.Event
+import com.github.se.eventradar.model.event.EventCategory
+import com.github.se.eventradar.model.event.EventTicket
+import com.github.se.eventradar.model.message.Message
+import com.github.se.eventradar.model.message.MessageHistory
+import com.github.se.eventradar.model.repository.event.IEventRepository
+import com.github.se.eventradar.model.repository.event.MockEventRepository
+import com.github.se.eventradar.model.repository.message.IMessageRepository
+import com.github.se.eventradar.model.repository.message.MockMessageRepository
+import com.github.se.eventradar.model.repository.user.IUserRepository
+import com.github.se.eventradar.model.repository.user.MockUserRepository
+import com.github.se.eventradar.screens.ChatScreen
+import com.github.se.eventradar.screens.EventDetailsScreen
+import com.github.se.eventradar.screens.HomeScreen
+import com.github.se.eventradar.screens.MessagesScreen
+import com.github.se.eventradar.screens.ViewFriendsProfileScreen
+import com.github.se.eventradar.ui.chat.ChatScreen
+import com.github.se.eventradar.ui.event.EventDetails
+import com.github.se.eventradar.ui.home.HomeScreen
+import com.github.se.eventradar.ui.messages.MessagesScreen
+import com.github.se.eventradar.ui.navigation.NavigationActions
+import com.github.se.eventradar.ui.navigation.Route
+import com.github.se.eventradar.ui.theme.MyApplicationTheme
+import com.github.se.eventradar.ui.viewProfile.ViewFriendsProfileUi
+import com.github.se.eventradar.viewmodel.ChatUiState
+import com.github.se.eventradar.viewmodel.ChatViewModel
+import com.github.se.eventradar.viewmodel.EventDetailsViewModel
+import com.github.se.eventradar.viewmodel.EventsOverviewViewModel
+import com.github.se.eventradar.viewmodel.MessagesViewModel
+import com.github.se.eventradar.viewmodel.ViewFriendsProfileUiState
+import com.github.se.eventradar.viewmodel.ViewFriendsProfileViewModel
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import io.github.kakaocup.compose.node.element.ComposeScreen
+import io.mockk.Runs
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.junit4.MockKRule
+import io.mockk.just
+import java.time.LocalDateTime
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class UserFlowTests : TestCase() {
+  @get:Rule val composeTestRule = createComposeRule()
+
+  // This rule automatically initializes lateinit properties with @MockK, @RelaxedMockK, etc.
+  @get:Rule val mockkRule = MockKRule(this)
+
+  // Relaxed mocks methods have a default implementation returning values
+  @RelaxedMockK lateinit var mockNavActions: NavigationActions
+
+  @RelaxedMockK lateinit var mockViewFriendsProfileViewModel: ViewFriendsProfileViewModel
```

Comment:
Why do you have to mock the VM?

Reply:
I only mocked hilt view models, following what was done in other test files.

Reply:
For integration tests there shouldn't need to be a mock. As long as the data repos are mocked the VMs should run as expected during the test